### PR TITLE
Improve client compatibility

### LINKED ISSUE #32: Add compatibility for prysm, nimbus and lodestar
## Summary

`eth-duties` currently only supports `teku` and `lighthouse` as connected beacon clients. This should be improved.

### PR DESCRIPTION
**Pull request by TobiWo**
/release

## Summary

Currently `eth-duties` only supports Teku and Lighthouse due to the fact the requests were not implemented following the official beacon api spec. This PR will add compatibility for:

* Nimbus
* Lodestar
* Prysm

## Note on Lodestar

I needed to add a specific change how requests are sent because Lodestar does not support URI percent-encoding currently. This was reported as [bug in the Lodestar github repository](https://github.com/ChainSafe/lodestar/issues/5403). If it will be added/fixed, I will adapt `eth-duties` as well (back to usage of percent-encoding for all requests).

### Update on Lodestar specifics
It turns out that Lodestar is the only client implementing percent-encoding correctly or using the right library for handling requests. The `,` is a reserved delimiting character which therefore should not be encoded. Fur further details please follow the discussion in the aforementioned issue in the Lodestar repository.

Closes #32

### CODE REVIEW

File: duties/constants/program.py
Comment (nflaig) [inline duties/constants/program.py:12]: the response should be json as well, I don't think any client returns plain text

```suggestion
REQUEST_HEADER = {"Content-type": "application/json", "Accept": "application/json"}
```

**Comment by TobiWo** [comment]
@nflaig Thanks for reviewing. I'm currently waiting for @LuisNaldo7 to finish the review on #40 . If this is done and I rebased the changes into this branch I would ask you to approve this PR which will then trigger a new release :raised_hands: .

**Comment by nflaig** [comment]
@TobiWo sounds good, will give it a final review after rebase

**Review by LuisNaldo7 — DISMISSED** [review]

**Comment by TobiWo** [comment]
@nflaig Sorry for the delay. This PR is now ready for review.

File: duties/protocol/request.py
Comment (nflaig) [inline duties/protocol/request.py:97]: this should probably be called `params` to make it a bit clearer

**Review by nflaig — APPROVED** [review]
lgtm

File: duties/protocol/request.py
Comment (nflaig) [reply duties/protocol/request.py]: that's even better 👍

Comment (TobiWo) [reply duties/protocol/request.py]: Then I'm looking forward that you close the conversation and approve the PR :smiley: . Will merge today in the evening (CET).

**Review by nflaig — APPROVED** [review]

File: duties/protocol/request.py
Comment (nflaig) [reply duties/protocol/request.py]: I am not able to resolve the conversation, but reviewed again and re-approved it 👍

**Review by LuisNaldo7 — APPROVED** [review]

**Comment by TobiWo** [comment]
@nflaig Unfortunately a review is only valid when you have write access. Therefore, if you like, I would add you as collaborator so you are able to do valid reviews in the future. Don't know if you have time for that?

**Comment by nflaig** [comment]
@TobiWo Thanks for the offer but I won't be able to find the time right now to familiarize myself with the code base to do proper reviews. That being said, feel free to just tag me on any PRs related to Lodestar or any other CL as well if you feel like my feedback would be useful.

**Pull request merged**
